src/engine/Server.h

```diff
@@ -161,10 +170,16 @@ class Server {
                                             SharedCancellationHandle handle,
                                             TimeLimit timeLimit) const;
 
-  ///
+  /// Acquire the `CancellationHandle` for the given `QueryId`, start the
+  /// watchdog and call `cancelAfterDeadline` to set the timeout after
+  /// `timeLimit`. Return an object of type
+  /// `CancellationHandleAndTimeoutTimerCancel`, where the `cancelTimeout_`
+  /// member can be invoked to cancel the imminent cancellation via timeout.
```

Comment:
I thought the `cancelTimeout_` member automatically cancels the timeout timer when it is destroyed? This could be made clear here.